src/Reflection/Annotations/AnnotationMethodReflection.php

```diff
@@ -109,6 +110,13 @@ public function getThrowType(): ?Type
 
 	public function hasSideEffects(): TrinaryLogic
 	{
+		if (
+			strtolower($this->getName()) !== '__construct'
```

Comment:
I think you can delete the thing about __construct.

Reply:
Removed